src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -571,6 +577,63 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 				});
 				widget.ariaLabel = ariaLabel;
 				widget.tabIndex = 0;
+			} else if (attachment.isImage) {
+				const blob = new Blob([attachment.value as Uint8Array], { type: 'image/png' });
+				const url = URL.createObjectURL(blob);
+
+				const img = document.createElement('img');
+				img.classList.add('chat-attached-context-image');
+				img.src = url;
+				img.alt = 'Image from Uint8Array';
+
+				// Pill with tiny image
+				const pill = document.createElement('div');
+				pill.classList.add('chat-attached-context-pill');
+				pill.tabIndex = -1;
+
+				const pillImg = document.createElement('img');
+				pillImg.src = url;
+				pillImg.alt = 'Image description';
+				pillImg.classList.add('chat-attached-context-pill-image');
+
+				pill.appendChild(pillImg);
+
+				// Custom label
+				const textLabel = document.createElement('span');
+				textLabel.textContent = localize('chat.imageLabel', '{0}', attachment.name);
+				textLabel.classList.add('chat-attached-context-custom-text');
+				textLabel.tabIndex = -1;
+
+
+				widget.style.position = 'relative';
+				widget.appendChild(pill);
+				widget.appendChild(textLabel);
+
+				const ariaLabel = localize('chat.imageAttachment', "Attached image, {0}", attachment.name);
+
+				// use icon instead of image pill
+				// const attachmentLabel = attachment.fullName ?? attachment.name;
+				// const withIcon = attachment.icon?.id ? `$(${attachment.icon.id}) ${attachmentLabel}` : attachmentLabel;
+				// label.setLabel(withIcon, undefined);
+
+				widget.ariaLabel = ariaLabel;
+				widget.tabIndex = 0;
+
+				// Hover
+				const hoverElement = document.createElement('div');
+				hoverElement.appendChild(img);
+				hoverElement.classList.add('chat-attached-context-hover');
+				hoverElement.setAttribute('aria-label', ariaLabel);
+
+				this._register(this.hoverService.setupManagedHover(getDefaultHoverDelegate('mouse'), widget, hoverElement));
+
+				// No delay for keyboard
+				widget.addEventListener('keydown', (event) => {
+					if (event.key === 'Enter' || event.key === ' ') {
+						this.hoverService.showManagedHover(widget);
+					}
```

Comment:
You can do this in a more strongly typed way via `StandardKeyboardEvent` https://github.com/microsoft/vscode/blob/6c478a6b044c36a48a3559213f9f8d0460831612/src/vs/base/browser/dom.ts#L2027-L2046